src/frontends/lean/parser.cpp

```diff
@@ -206,7 +206,7 @@ void parser::check_break_at_pos(break_at_pos_exception::token_context ctxt) {
 void parser::check_break_before(break_at_pos_exception::token_context ctxt) {
     if (!get_complete())
         ctxt = break_at_pos_exception::token_context::none;
-    if (m_break_at_pos && *m_break_at_pos < pos())
+    if (m_break_at_pos && *m_break_at_pos < end_pos())
```

Comment:
I'm currently porting some mathlib code to work with current master + this, and I may revert this change - this breaks working with half-completed tactic states in quite a harsh way.